pages/agent/v3/signed_pipelines.md

```diff
@@ -45,86 +47,117 @@ Additionally, we also sign the following aspects of the build for each job:
 
-> Some older Buildkite pipelines don't use YAML for their configured steps - these can't be signed, and will have to be migrated to use the new format to be used with signed pipelines - see [Migrating to YAML steps](/docs/tutorials/pipeline_upgrade)
+> 🚧 Non-YAML steps
+> You must use YAML to sign steps configured in the Buildkite dashboard. If you don't use YAML, you'll need to [migrate to YAML steps](/docs/tutorials/pipeline_upgrade) before continuing.
 
-You may have noticed that when we configured the signing instances, we gave them the capability to both sign and verify jobs. For maximum security, the configured steps - usually a single step that uploads (and signs!) the remainder of the pipeline - should themselves be signed.
+To sign steps configured in the Buildkite dashboard, you need to add static signatures to the YAML:
 
-The solution here is to add static signatures to the steps in your pipeline's configured steps. To do this, copy the YAML from the editor on the pipeline's page in Buildkite, and paste it into a file on your machine. Then, run:
+1. Copy the existing YAML steps definition from the editor in the Builkdite dashboard.
+1. Paste the step definitions into a file on your machine.
+1. Generate a new YAML definition with signatures by running:
 
-<!-- We're going to change these commands at some point soon -->
+    ```sh
+    buildkite-agent pipeline upload --jwks-file-path <path-to-private-keys> --signing-key-id <key-id> --dry-run --format yaml <path-to-pipeline-file>
+    ```
 
-```bash
-buildkite-agent pipeline upload --jwks-file-path <path to private keys> --signing-key-id <key id> --dry-run --format yaml <path to pipeline file>
-```
+    Replacing the following:
+    * `<path-to-private-keys>` with the path to your private keys.
+    * `<key-id>` with the key ID.
+    * `<path-to-pipeline-file>` with the path to the step definitions you saved from the Buildkite dashboard.
+
+    This outputs the pipeline YAML with the signatures added to the steps.
+
+1. Copy the updated step definitions back into the editor in the Buildkite dashboard.
+1. Save the new definition.
 
-For example, if you had a file called `pipeline.yaml` in the current directory, and you wanted to sign it with the key we generated earlier, you'd run:
+For example, if you had a file called `pipeline.yaml` in the current directory, and you wanted to sign it with the key generated earlier, you'd run:
 
 ```bash
-buildkite-agent pipeline upload --jwks-file-path /path/to/private/key.json --signing-key-id my-key-id --dry-run --format yaml configured-steps.yaml
+buildkite-agent pipeline upload --jwks-file-path /path/to/private/key.json --signing-key-id my-key-id --dry-run --format yaml pipeline.yaml
```

Comment:
potentially we should switch it so that they're both the same and _not_ `pipeline.yaml`, eg

```suggestion
For example, if you had a file called `configured-steps.yaml` in the current directory, and you wanted to sign it with the key generated earlier, you'd run:
```bash
buildkite-agent pipeline upload --jwks-file-path /path/to/private/key.json --signing-key-id my-key-id --dry-run --format yaml configured-steps.yaml
```

to avoid confusion with the _actual_ pipeline.yaml, which is getting signed when it gets uploaded 😅 there's a lot of overlapping nomenclature here...

potentially `configured-steps.yaml` no longer makes sense as we're not introducing that term anymore, but i think it should be something that isn't `pipeline.yaml`